docs/user/container.md

```diff
@@ -231,41 +246,129 @@ Access Control List   A::OWNER@:rwdtTaAo, A:G:GROUP@:rwtT
-| acl                     | No              | Container access control list.|
-| compression             | Yes             | Whether online compression is enabled (off, lz4, deflate[1-4])|
-| dedup                   | Yes             | Inline deduplication off, hash based (hash) or using memory compare (memcmp)|
+| acl                     | No              | Container access control list|
+| layout\_type            | Yes             | Container type (e.g., POSIX, HDF5, ...)|
+| layout\_ver             | Yes             | Layout version to be used at the discretion of I/O middleware for interoperability|
+| rf                      | Yes             | Redundancy Factor which is the maximum number of simultaneous engine failures that objects can support without data loss|
+| rf\_lvl                 | Yes             | Redudancy Factor Level which is the level in the fault domain hierachy to use for object placement|
+| health                  | No              | Current state of the container|
+| alloc\_oid              | No              | Maximum allocated object ID by container allocator|
+| ec\_cell                | Yes             | Erasure code cell size for erasure-coded objects|
+| cksum                   | Yes             | Checksum off, or algorithm to use (adler32, crc[16,32,64] or sha[1,256,512])|
+| cksum\_size             | Yes             | Checksum Size determining the maximum extent size that a checksum can cover|
+| srv\_cksum              | Yes             | Whether to verify checksum on the server before writting data (default: off)|
+
+
+Moreover, the following properties have been added as placeholders, but are not
+fully supported yet:
+
+| **Container Property**  | **Immutable**   | **Description** |
+| ------------------------| --------------- | ----------------|
+| max\_snapshot           | No              | Impose a upper limit on number of snapshots to retain (default: 0, no limitation)|
+| compression             | Yes             | Online compression off, or algorithm to use (off, lz4, deflate[1-4])|
+| dedup                   | Yes             | Inline deduplication off, or algorithm to use (hash or memcmp)|
 | dedup\_threshold        | Yes             | Minimum I/O size to consider for deduplication|
 | encryption              | Yes             | Inline encryption off, or algorithm to use (XTS[128,256], CBC[128,192,256] or GCM[128,256])|
-| status                  | No              | Current state of the container|
-| alloc\_oid              | No              | Maximum allocated object ID by container allocator|
 
-Refer to the [Data Integrity](#Data-Integrity) and [Access Control Lists](#Access_Control_Lists)
-sections for more details on the checksum and access-related properties.
+Please refer to the next sections for more details on each property.
+
+### Redundancy Factor
+
+Different objects in a DAOS container may belong to different object classes and
```

Comment:
Maybe:
Different objects in a DAOS container may belong to different object classes and have different levels of data protection. While this model gives a lot of control to the users, it also requires carefully selecting a suitable class for each object. If objects with different data protection levels are stored in the same container, the user should also be prepared for the case where some objects might suffer from data loss after several cascading failures, while some others with a higher level of data protection may not. This incurs extra complexity that not all I/O middleware necessarily wants to deal with.

To lower the bar of adoption while still keeping the flexibility, two properties have been introduced:

-	The redundancy factor (rf) describes the number of concurrent engine exclusions that objects in the container are protected against. The rf value is an integer between 0 (no data protection) and 5 (support up to 5 simultaneous failures).
-	a `health` property representing whether any object content might have been lost due to cascading engine failures. The value of this property can be either `HEALTHY` (no data loss) or `UNCLEAN` (data might have been lost).

The redundancy factor can be set at container creation time and cannot be modified after creation.